spec/v1/envelope.md

```diff
@@ -19,5 +21,41 @@ standard DSSE fields.
     specifying its schema.
 -   `payload` MUST be a base64-encoded JSON [Statement].
 
+## File naming convention
+
+Envelopes SHOULD use the suffix `.json`.
+
+-   An Envelope containing an attestation about a particular SW supply chain
+    step `<step-name>` SHOULD be named `<step-name>.json`.
+-   If multiple Envelopes are produced for the same step by different
```

Comment:
Some thoughts:
1. Currently the KeyID is included even if there's a single envelope for the step. I think it's okay to loosen that requirement but I'm noting it for completeness.
2. The pubkey hash is not well defined. [The spec](https://github.com/in-toto/docs/blob/master/in-toto-spec.md#421-key-formats) says "The KEYID of a key is the hexadecimal encoding of the SHA-256 hash of the canonical JSON form of the public key." This is not the same as "pubkey-hash" as the key ID is calculated across the key type, scheme, etc. fields as well, _including some that are not documented in the spec_.
3. Existing implementations actually use a [shortened version](https://github.com/in-toto/in-toto/blob/develop/in_toto/models/link.py#L30) of the full key ID.
4. I wonder if we also want to consider moving away from the strict key ID requirements, something like TUF's [TAP-12](https://github.com/theupdateframework/taps/blob/master/tap12.md).

Reply:
Oh I missed that part of the spec, thanks for pointing that out. For the sake of moving this PR along, I would either go with options 3 or 4 where we match the current implementation (KeyID[0:8]) or TAP-12.

Reply:
I recommend 3 for consistency until we can sort it out across both specs.

Reply:
That was my inclination, too. I'll make those updates